star-dash/star-dash/GameEngine/Systems/System.swift

```diff
@@ -0,0 +1,20 @@
+//
+//  System.swift
+//  star-dash
+//
+//  Created by Ho Jun Hao on 13/3/24.
+//
+
+import Foundation
+
+protocol System {
+    var isActive: Bool { get set }
+    var dispatcher: EventModifiable { get set }
+    var entityManager: EntityManager { get set }
+
+    func update(by deltaTime: TimeInterval)
```

Comment:
Can I clarify what is the role of this method? I'm a bit confused because I noticed that not all systems (like `PositionSystem`) override this method. And in `PhysicsSystem`, the `update` method basically sets all the force vectors to 0. Under what circumstance is this method called?

Reply:
It more like for timed components in the game, e.g. those that after a certain time will stop working like powerups